plugins/reachability/src/main/java/org/qbicc/plugin/reachability/RTAInfo.java

```diff
@@ -360,10 +379,49 @@ synchronized void processInstantiatedClass(final LoadedTypeDefinition type, bool
         }
     }
 
+    synchronized void processServiceInterface(final LoadedTypeDefinition ltd, final String internalName) {
+        if (ltd.isInterface() && !isServiceSearched(ltd)) {
+            /* process service interface */
+            addReachableInterface(ltd);
+
+            /* find and load service implementations */
+            String fileName = "META-INF/services/" + internalName.replace("/", ".");
+            try {
+                Enumeration<URL> servicefiles = ClassLoader.getSystemResources(fileName);
+                while (servicefiles.hasMoreElements()) {
+                    URL file = servicefiles.nextElement();
+                    BufferedReader in = new BufferedReader(new InputStreamReader(file.openStream()));
+
+                    String line;
+                    while ((line = in.readLine()) != null) {
+                        if (!line.isBlank()) {
+                            DefinedTypeDefinition serviceImplDefinition = ctxt.getBootstrapClassContext().findDefinedType(line.replace(".", "/"));
+                            LoadedTypeDefinition serviceImpl = serviceImplDefinition.load();
+                            addReachableClass(serviceImpl);
```

Comment:
I suspect that `org.qbicc.plugin.reachability.RTAInfo#processInstantiatedClass` might be more appropriate for this.  The service loader would instantiate the class, but that does not make it reachable unless the result of the service load was used.